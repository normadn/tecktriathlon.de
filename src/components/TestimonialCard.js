import React from "react";

import quotes from "../assets/img/icons/quotesIcon.svg";

const TestimonialCard = (props) => (
  <div
    css={{
      display: props.mobile ? "block" : "inline-block",
      height: props.mobile ? null : "375px",
      whiteSpace: "normal",
      width: props.mobile ? null : "530px",
      borderRadius: "8px",
      boxShadow: "4px 7px 20px 2px rgba(7, 16, 75, 0.1)",
      border: "solid 1px #e8e8e8",
      padding: props.mobile ? "24px 24px 82px 24px" : "48px",
      margin: props.mobile ? "45px 20px" : "45px 23px",
      flex: props.mobile ? null : "0 0 530px",
      backgroundColor: "white",
      zIndex: 2,
      position: "relative",
      opacity: props.mobile || !props.index ? 1 : 0.6,
      transition: !props.index
        ? "opacity 300ms ease 0ms"
        : "opacity 300ms ease 200ms"
    }}
  >
    <div
      css={{
        display: "flex",
        alignItems: "center",
        marginBottom: "20px"
      }}
    >
      <img
        src={props.item.image}
        css={{ borderRadius: "50%", height: "72px", width: "72px" }}
        alt=""
      />
      <div css={{ display: "inline-block", marginLeft: "24px" }}>
        <strong
          css={{
            marginBottom: "8px",
            fontSize: "20px",
            display: "block"
          }}
        >
          {props.item.name}
        </strong>
        {props.item.project ? (
          <strong>
            <span css={{ color: "#838383", fontWeight: 600 }}>Firma:</span>{" "}
            <a
              rel="external noopener"
              target="_blank"
              href={props.item.projectLink}
              css={{ fontWeight: 600 }}
            >
              {props.item.project}
            </a>
          </strong>
        ) : (
          ""
        )}
      </div>
    </div>
    <p
      css={{
        lineHeight: "32px",
        fontSize: "15px"
      }}
    >
      {props.item.text}
    </p>
    <img
      src={quotes}
      css={{
        position: "absolute",
        bottom: 24,
        right: 40
      }}
      alt=""
    />
  </div>
);

export default TestimonialCard;
