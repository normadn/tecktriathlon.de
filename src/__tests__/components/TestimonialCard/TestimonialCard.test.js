import React from "react";
import { shallow } from "enzyme";

import TestimonialCard from "../../../components/TestimonialCard";

import placeholderHeadshot from "../../../assets/img/testimonials/placeholderHeadshot.jpg";

describe("TestimonialCard0", () => {
  it("renders correctly", () => {
    expect(
      shallow(
        <TestimonialCard
          item={{
            name: "Ross Hill",
            project: "Dolores",
            image: placeholderHeadshot,
            text:
              "Bring to the table win-win survival strategies to ensure proactive domination. At the end of the day, going forward, a new normal that has evolved from generation X is on the runway heading towards a streamlined cloud solution. User generated content in real-time will have multiple touchpoints for offshoring."
          }}
          index={0}
        />
      )
    ).toMatchSnapshot();
  });
});

describe("TestimonialCard1", () => {
  it("renders correctly", () => {
    expect(
      shallow(
        <TestimonialCard
          item={{
            name: "Abc Xyz",
            project: "Test1",
            image: placeholderHeadshot,
            text:
              "Bring to the table win-win survival strategies to ensure proactive domination. At the end of the day, going forward, a new normal that has evolved from generation X is on the runway heading towards a streamlined cloud solution. User generated content in real-time will have multiple touchpoints for offshoring."
          }}
          index={1}
        />
      )
    ).toMatchSnapshot();
  });
});
