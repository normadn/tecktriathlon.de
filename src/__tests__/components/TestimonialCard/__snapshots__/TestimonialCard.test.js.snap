// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestimonialCard0 renders correctly 1`] = `
<div
  css={
    Object {
      "backgroundColor": "white",
      "border": "solid 1px #e8e8e8",
      "borderRadius": "8px",
      "boxShadow": "4px 7px 20px 2px rgba(7, 16, 75, 0.1)",
      "display": "inline-block",
      "flex": "0 0 530px",
      "height": "375px",
      "margin": "45px 23px",
      "opacity": 1,
      "padding": "48px",
      "position": "relative",
      "transition": "opacity 300ms ease 0ms",
      "whiteSpace": "normal",
      "width": "530px",
      "zIndex": 2,
    }
  }
>
  <div
    css={
      Object {
        "alignItems": "center",
        "display": "flex",
        "marginBottom": "20px",
      }
    }
  >
    <img
      alt=""
      css={
        Object {
          "borderRadius": "50%",
          "height": "72px",
          "width": "72px",
        }
      }
      src="test-file-stub"
    />
    <div
      css={
        Object {
          "display": "inline-block",
          "marginLeft": "24px",
        }
      }
    >
      <strong
        css={
          Object {
            "display": "block",
            "fontSize": "20px",
            "marginBottom": "8px",
          }
        }
      >
        Ross Hill
      </strong>
      <strong>
        <span
          css={
            Object {
              "color": "#838383",
              "fontWeight": 600,
            }
          }
        >
          Project:
        </span>
         
        <a
          css={
            Object {
              "fontWeight": 600,
            }
          }
          rel="external noopener"
          target="_blank"
        >
          Dolores
        </a>
      </strong>
    </div>
  </div>
  <p
    css={
      Object {
        "fontSize": "15px",
        "lineHeight": "32px",
      }
    }
  >
    Bring to the table win-win survival strategies to ensure proactive domination. At the end of the day, going forward, a new normal that has evolved from generation X is on the runway heading towards a streamlined cloud solution. User generated content in real-time will have multiple touchpoints for offshoring.
  </p>
  <img
    alt=""
    css={
      Object {
        "bottom": 24,
        "position": "absolute",
        "right": 40,
      }
    }
    src="test-file-stub"
  />
</div>
`;

exports[`TestimonialCard1 renders correctly 1`] = `
<div
  css={
    Object {
      "backgroundColor": "white",
      "border": "solid 1px #e8e8e8",
      "borderRadius": "8px",
      "boxShadow": "4px 7px 20px 2px rgba(7, 16, 75, 0.1)",
      "display": "inline-block",
      "flex": "0 0 530px",
      "height": "375px",
      "margin": "45px 23px",
      "opacity": 0.6,
      "padding": "48px",
      "position": "relative",
      "transition": "opacity 300ms ease 200ms",
      "whiteSpace": "normal",
      "width": "530px",
      "zIndex": 2,
    }
  }
>
  <div
    css={
      Object {
        "alignItems": "center",
        "display": "flex",
        "marginBottom": "20px",
      }
    }
  >
    <img
      alt=""
      css={
        Object {
          "borderRadius": "50%",
          "height": "72px",
          "width": "72px",
        }
      }
      src="test-file-stub"
    />
    <div
      css={
        Object {
          "display": "inline-block",
          "marginLeft": "24px",
        }
      }
    >
      <strong
        css={
          Object {
            "display": "block",
            "fontSize": "20px",
            "marginBottom": "8px",
          }
        }
      >
        Abc Xyz
      </strong>
      <strong>
        <span
          css={
            Object {
              "color": "#838383",
              "fontWeight": 600,
            }
          }
        >
          Project:
        </span>
         
        <a
          css={
            Object {
              "fontWeight": 600,
            }
          }
          rel="external noopener"
          target="_blank"
        >
          Test1
        </a>
      </strong>
    </div>
  </div>
  <p
    css={
      Object {
        "fontSize": "15px",
        "lineHeight": "32px",
      }
    }
  >
    Bring to the table win-win survival strategies to ensure proactive domination. At the end of the day, going forward, a new normal that has evolved from generation X is on the runway heading towards a streamlined cloud solution. User generated content in real-time will have multiple touchpoints for offshoring.
  </p>
  <img
    alt=""
    css={
      Object {
        "bottom": 24,
        "position": "absolute",
        "right": 40,
      }
    }
    src="test-file-stub"
  />
</div>
`;
